include/devicetree.h

```diff
@@ -467,6 +467,36 @@
  */
 #define DT_NODE_FULL_NAME(node_id) DT_CAT(node_id, _FULL_NAME)
 
+/**
+ * @brief Get the linker memory-region name
+ *
+ * This attempts to use the zephyr,linker-region property, falling back
+ * to the node path if it doesn't exist.
+ *
+ * Example devicetree fragment:
+ *
+ *     / {
+ *             soc {
+ *                     sram1: memory@2000000 {
+ *                         zephyr,linker-region = "MY_NAME";
+ *                     };
+ *                     sram2: memory@2001000 { ... };
+ *             };
+ *     };
+ *
+ * Example usage:
+ *
+ *    DT_NODE_LINKER_REGION(DT_NODELABEL(sram1)) // "MY_NAME"
+ *    DT_NODE_LINKER_REGION(DT_NODELABEL(sram2)) // "/soc/memory@2001000"
+ *
+ * @param node_id node identifier
+ * @return the name of the linker region the node will generate
+ */
+#define DT_NODE_LINKER_REGION(node_id)				     \
```

Comment:
put this in its own header.  Would like to keep the devicetree.h to just generic functionality.

Reply:
> or does putting it in `include/linker/devicetree_regions.h ` make sense like how we do `GPIO_DT_SPEC_GET_BY_IDX` in `include/drivers/gpio.h`

I like that.

Reply:
Macro move to `include/linker/devicetree_regions.h`.